Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -420,11 +418,55 @@ private int PreOrderChecks(string symbol, int quantity, OrderType type)
                     // tell the user we require an 11 minute buffer, on minute data in live a user will receive the 3:49->3:50 bar at 3:50,
                     // this is already too late to submit one of these orders, so make the user do it at the 3:48->3:49 bar so it's submitted
                     // to the brokerage before 3:50.
-                    Error("MarketOnClose orders must be placed with at least a 11 minute buffer before market close.");
-                    return -6;
+                    response.Error(OrderResponseErrorCode.MarketOnCloseOrderTooLate, "MarketOnClose orders must be placed with at least a 11 minute buffer before market close.");
+                    Error(response.ErrorMessage);
+                    return response;
                 }
             }
-            return 0;
+
+            response.Processed();
+
+            return response;
+        }
+
+        /// <summary>
+        /// Cancel order
+        /// </summary>
+        /// <param name="orderId"></param>
+        /// <returns></returns>
+        public OrderResponse CancelOrder(int orderId)
+        {
+            return Transactions.CancelOrder(orderId);
+        }
+
+        public OrderResponse UpdateOrder(MarketOrder order, int? quantity = null, string tag = null)
+        {
+            return Transactions.UpdateOrder(order.UpdateRequest(quantity, tag));
+        }
+
+        public OrderResponse UpdateOrder(MarketOnOpenOrder order, int? quantity = null, string tag = null)
+        {
+            return Transactions.UpdateOrder(order.UpdateRequest(quantity, tag));
+        }
+
+        public OrderResponse UpdateOrder(MarketOnCloseOrder order, int? quantity = null, string tag = null)
+        {
+            return Transactions.UpdateOrder(order.UpdateRequest(quantity, tag));
+        }
+
+        public OrderResponse UpdateOrder(LimitOrder order, int? quantity = null, decimal? limitPrice = null, string tag = null)
+        {
+            return Transactions.UpdateOrder(order.UpdateRequest(quantity, limitPrice, tag));
+        }
+
+        public OrderResponse UpdateOrder(StopLimitOrder order, int? quantity = null, decimal? stopPrice = null, decimal? limitPrice = null, string tag = null)
+        {
+            return Transactions.UpdateOrder(order.UpdateRequest(quantity, stopPrice, limitPrice, tag));
+        }
+
+        public OrderResponse UpdateOrder(StopMarketOrder order, int? quantity = null, decimal? stopPrice = null, string tag = null)
```

Comment:
Let's modify the signatures of these `UpdateOrder` functions to move the price parameters first. In most cases, users will be updating the price of a limit/stop order vs the quantity of the order.
```C#
        public OrderResponse UpdateOrder(LimitOrder order, decimal? limitPrice = null, int? quantity = null, string tag = null)
        {
            return Transactions.UpdateOrder(order.UpdateRequest(quantity, limitPrice, tag));
        }

        public OrderResponse UpdateOrder(StopLimitOrder order, decimal? stopPrice = null, decimal? limitPrice = null, int? quantity = null, string tag = null)
        {
            return Transactions.UpdateOrder(order.UpdateRequest(quantity, stopPrice, limitPrice, tag));
        }

        public OrderResponse UpdateOrder(StopMarketOrder order, decimal? stopPrice = null, int? quantity = null, string tag = null)
        {
            return Transactions.UpdateOrder(order.UpdateRequest(quantity, stopPrice, tag));
        }
```